designcompose/src/main/java/com/android/designcompose/Utils.kt

```diff
@@ -590,6 +590,138 @@ internal fun mergeStyles(base: ViewStyle, override: ViewStyle): ViewStyle {
+    builder.text_align_vertical = text_align_vertical
+    builder.text_overflow = text_overflow
+    builder.text_shadow = text_shadow
+    builder.node_size = node_size
+    builder.line_height = line_height
+    builder.line_count = line_count
+    builder.font_features = font_features
+    builder.filter = filter
+    builder.backdrop_filter = backdrop_filter
+    builder.blend_mode = blend_mode
+    // We don't need to copy any of the layout properties, but do it anyway to
+    // avoid confusion if this function later gets used more generally; we should
+    // refactor ViewStyle to separate the fields used for rendering from those used
+    // for layout.
+    builder.display_type = display_type
+    builder.position_type = position_type
+    builder.flex_direction = flex_direction
+    builder.flex_wrap = flex_wrap
+    builder.grid_layout = grid_layout
+    builder.grid_columns_rows = grid_columns_rows
+    builder.grid_adaptive_min_size = grid_adaptive_min_size
+    builder.grid_span_content = grid_span_content
+    builder.overflow = overflow
+    builder.max_children = max_children
+    builder.overflow_node_id = overflow_node_id
+    builder.overflow_node_name = overflow_node_name
+    builder.align_items = align_items
+    builder.align_self = align_self
+    builder.align_content = align_content
+    builder.justify_content = justify_content
+    builder.top = top
+    builder.left = left
+    builder.bottom = bottom
+    builder.right = right
+    builder.margin = margin
+    builder.padding = padding
+    builder.item_spacing = item_spacing
+    builder.cross_axis_item_spacing = cross_axis_item_spacing
+    builder.flex_grow = flex_grow
+    builder.flex_shrink = flex_shrink
+    builder.flex_basis = flex_basis
+    builder.bounding_box = bounding_box
+    builder.horizontal_sizing = horizontal_sizing
+    builder.vertical_sizing = vertical_sizing
+    builder.width = width
+    builder.height = height
+    builder.min_width = min_width
+    builder.min_height = min_height
+    builder.max_width = max_width
+    builder.max_height = max_height
+    builder.aspect_ratio = aspect_ratio
+    builder.pointer_events = pointer_events
+    builder.meter_data = meter_data
+    return builder
+}
+
+// Take the external layout fields of ViewStyle and put them into an ExternalLayoutData object.
+internal fun ViewStyle.externalLayoutData(): ExternalLayoutData {
+    return ExternalLayoutData(
+        margin,
+        top,
+        left,
+        bottom,
+        right,
+        width,
+        height,
+        min_width,
+        min_height,
+        max_width,
+        max_height,
+        node_size,
+        bounding_box,
+        flex_grow,
+        flex_basis,
+        align_self,
+        position_type,
+        transform,
+        relative_transform,
+    )
+}
+
+// Take the external layout data and merge it into this ViewStyle, overriding its values and
+// returning a new ViewStyle.
+internal fun ViewStyle.withExternalLayoutData(
```

Comment:
It might make sense to make this a method on the builder instead of on `ViewStyle` directly (but that would make the call site a bit noisier). I don't think it's urgent.

Reply:
I'll consider this in a follow up.